cmd/main.go

```diff
@@ -1,15 +1,24 @@
 package main
 
 import (
+	"github.com/cyneptic/letsgo/infrastructure/provider"
+	"github.com/labstack/echo/v4"
 	"log"
 
-	repositories "github.com/cyneptic/letsgo/internal/infrastructure/repository"
-	"github.com/labstack/echo/v4"
+	"github.com/cyneptic/letsgo/controller"
+	repositories "github.com/cyneptic/letsgo/infrastructure/repository"
+	"github.com/cyneptic/letsgo/internal/core/service"
+	"github.com/joho/godotenv"
 )
 
 func main() {
-	_ = repositories.NewGormDatabase()
+	_ = godotenv.Load(".env")
+	gormDb := repositories.NewGormDatabase()
+	redisDb := repositories.RedisInit()
+	paymentGateway := provider.NewMellatGateway()
+	srvPayment := service.NewPaymentService(redisDb, gormDb, paymentGateway)
```

Comment:
db, redis and provider creation should be done inside NewPaymentService() function.
and service should be made inside NewPaymentHandler() function.

this cleans up the main file and separates concerns giving the responsibility of making the dependencies of each to their constructor function.